opentelemetry/src/trace/mod.rs

```diff
@@ -302,11 +302,24 @@ pub struct Link {
 }
 
 impl Link {
-    /// Create a new link.
-    pub fn new(span_context: SpanContext, attributes: Vec<KeyValue>) -> Self {
+    /// Create new `Link`
+    pub fn new(
+        span_context: SpanContext,
+        attributes: Vec<KeyValue>,
+        dropped_attributes_count: u32,
```

Comment:
not really introduced in this PR, but the API exposing things like `dropped_attributes_count` is not good :( It is up-to the SDK to enforce the limit and store the count of attributes, not that of API.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute-limits

Reply:
@lalitb You don't plan to address this in this PR right?

Reply:
This needs to be addressed for both Span Events and Span Links. Let me address that separately. Will create an issue for that.

Reply:
@lalitb

Reply:
Done now - #1629